src/game/Entities/Player.cpp

```diff
@@ -7677,12 +7683,22 @@ void Player::SendInitWorldStates(uint32 zoneid) const
 
     switch (zoneid)
     {
-        case 139:                                       // Eastern Plaguelands
-        case 1377:                                      // Silithus
+    case 139:                                       // Eastern Plaguelands
+        if (sWorld.getConfig(CONFIG_BOOL_OUTDOORPVP_EP_ENABLED))
+        {
             if (OutdoorPvP* outdoorPvP = sOutdoorPvPMgr.GetScript(zoneid))
                 outdoorPvP->FillInitialWorldStates(data, count);
             break;
-    }
+        }
+            
+    case 1377:                                      // Silithus
+        if (sWorld.getConfig(CONFIG_BOOL_OUTDOORPVP_SI_ENABLED))
+        {
+            if (OutdoorPvP* outdoorPvP = sOutdoorPvPMgr.GetScript(zoneid))
+                outdoorPvP->FillInitialWorldStates(data, count);
```

Comment:
This shouldn't be necessary because the config option would not load a script for the zoneid, so GetScript(zoneid) returns nullptr for every zone that's disabled by config